pywt/_extensions/c/cwt.template.c

```diff
@@ -0,0 +1,343 @@
+/* Copyright (c) 2016 Holger Nahrstaedt */
+/* See COPYING for license details. */
+
+#include "templating.h"
+
+#ifndef TYPE
+#error TYPE must be defined here.
+#else
+
+#include "cwt.h"
+#include "convolution.h"
+
+#if defined _MSC_VER
+#define restrict __restrict
+#elif defined __GNUC__
+#define restrict __restrict__
+#endif
+#define _USE_MATH_DEFINES
+#include "math.h"
+
+TYPE CAT(TYPE, _powof)(const TYPE x, const TYPE y)
+{
+    if (sizeof(TYPE) == sizeof(double))
+        return pow(x,y);        
+    else
+        return powf(x,y);
+}
+
+
+
+int CAT(TYPE, _gaus)(const TYPE * const restrict input,
+                              TYPE * const restrict output, const size_t N,
+                              const size_t number){
+    //if (strcmp(wavelet->short_name, "gauss")){
+           switch (number) {
+               case 1:
+                    CAT(TYPE, _gaus1)(input, output, N);
```

Comment:
I would inline the calculation, and let the compiler unswitch it. So like:

```
for (i = 0; i < N; i++) {
  switch(number) {
    ...
  }
}
```

This should make things a lot more compact and readable, and any sensible compiler should optimize this correctly.